registration-processor/registration-processor-common-camel-bridge/src/main/java/io/mosip/registration/processor/camel/bridge/intercepter/PauseFlowPredicate.java

```diff
@@ -53,56 +58,85 @@ private void init() {
+        MessageDTO messageDto=objectMapper.readValue(message, MessageDTO.class);
 		LOGGER.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
 				"exchange.getFromEndpoint().toString() " + exchange.getFromEndpoint().toString());
 
 		String fromAddress = exchange.getFromEndpoint().toString();
+		WorkflowInternalActionDTO workflowInternalActionDTO = new WorkflowInternalActionDTO();
+		List<String> matchedRuleIds=new ArrayList<String>();
+		String ruleDescription="";
+		long pauseFor=0;
+		String defaultResumeAction=null;
+		Map<String,String> tags=messageDto.getTags();
 		for (Setting setting : settings) {
-			try {
+			if(isRuleIdNotPresent(tags,setting.getRuleId())) {
+
 			JSONArray jsonArray = JsonPath.read(message, setting.getMatchExpression());
 			if (Pattern.matches(setting.getFromAddress(), fromAddress)
 					&& !jsonArray.isEmpty()) {
-					WorkflowInternalActionDTO workflowInternalActionDTO = new WorkflowInternalActionDTO();
-					workflowInternalActionDTO.setResumeTimestamp(DateUtils
-						.formatToISOString(DateUtils.getUTCCurrentDateTime().plusSeconds(setting.getPauseFor())));
-					workflowInternalActionDTO.setRid(json.getString(JsonConstant.RID));
-					workflowInternalActionDTO.setDefaultResumeAction(setting.getDefaultResumeAction());
-					workflowInternalActionDTO.setActionCode(WorkflowInternalActionCode.MARK_AS_PAUSED.toString());
-					workflowInternalActionDTO
-							.setEventTimestamp(DateUtils.formatToISOString(DateUtils.getUTCCurrentDateTime()));
-					workflowInternalActionDTO
-							.setActionMessage(PlatformSuccessMessages.PACKET_MARK_AS_PAUSED.getMessage());
-					workflowInternalActionDTO.setResumeRemoveTags(setting.getResumeRemoveTags());
-					workflowInternalActionDTO.setReg_type(json.getString(JsonConstant.REGTYPE));
-					workflowInternalActionDTO.setIteration(json.getInteger(JsonConstant.ITERATION));
-					workflowInternalActionDTO.setSource(json.getString(JsonConstant.SOURCE));
-					workflowInternalActionDTO
-							.setWorkflowInstanceId(json.getString(JsonConstant.WORKFLOW_INSTANCE_ID));
-					exchange.getMessage().setBody(objectMapper.writeValueAsString(workflowInternalActionDTO));
-					return true;
-				} 
-			}catch (JsonProcessingException e) {
-				LOGGER.error("Error in  RoutePredicate::matches {}",
-						 e.getMessage());
-				throw new BaseUncheckedException(e.getMessage());
-				}
-				catch (InvalidPathException e) {
-				LOGGER.error("Error in  RoutePredicate::matches {}",
-							 e.getMessage());
-				throw new BaseUncheckedException(e.getMessage());
+				     matchedRuleIds.add(setting.getRuleId());
+				     if(ruleDescription.isBlank())
+				    	 ruleDescription=setting.getRuleDescription();
+				     else
+				    	 ruleDescription=ruleDescription+","+setting.getRuleDescription();
+				     if(setting.getPauseFor()>pauseFor) {
+				    	 pauseFor=setting.getPauseFor();
+				    	 defaultResumeAction=setting.getDefaultResumeAction();
+				     }

```

Comment:
Please add few unit test case to the new logic like
1. matched rule ids to be concatinated
2. pauseFor should be take for max and the respective defualt resume action to be taken 
3. rulesIds in the PAUSE_IMMUNITY_RULE_IDS tag to be ignored while considering settings

Reply:
Fixed and added  1 and 2 in testRoutePredicatePositive
3 in testRoutePredicatePositiveWithRuleIdsIgnore